packages/pwa/app/utils/utils.js

```diff
@@ -162,3 +163,153 @@ export const capitalize = (text) => {
+    }
+
+    const {hostname} = new URL(`${getAppOrigin()}${path}`)
+    site = getSiteByHostname(hostname)
+
+    if (site) {
+        return site
+    }
+
+    site = sitesConfig.find((site) => site.id === defaultSiteId)
+
+    return site
+}
+
+/**
+ * Get the site Id based on the given hostname
+ * @param {string} hostname
+ * @returns {string} siteId
+ */
+export const getSiteByHostname = (hostname) => {
+    const {
+        app: {sites: sitesConfig}
+    } = getConfig()
+    if (!sitesConfig.length) throw new Error('No site config found. Please check you configuration')
+    if (!hostname) return undefined
+
+    const site = sitesConfig.filter((site) => {
+        return site?.hostnames?.some((i) => i.includes(hostname))
+    })
+
+    return site?.length === 1 ? site[0] : undefined
+}
+
+/**
+ * return l10n config for current site
+ * @param url
+ * @returns {object}
+ */
+export const getL10nConfig = (url) => {
+    const sitesConfig = getSitesConfig()
+    if (!sitesConfig.length) throw new Error('No site config found. Please check you configuration')
+    const siteId = getSite(url)?.id
+    const l10nConfig = sitesConfig.find((site) => site.id === siteId)?.l10n
+    return l10nConfig
+}
+
+export const getSiteByUrl = (url) => {
+    let path = url
+    if (!path) {
+        path = `${window?.location.pathname}${window?.location.search}`
+    }
+    const {siteIdsRegExp, siteAliasesRegExp} = getSitesRegExp()
+    let site
+    const aliasMatch = path.match(siteAliasesRegExp)
+    const idMatch = path.match(siteIdsRegExp)
+    if (aliasMatch) {
+        // clean up any non-character
+        const siteAlias = aliasMatch[0].replace(/\W/g, '')
+        site = getSiteByAlias(siteAlias)
+    } else if (idMatch) {
+        const siteId = idMatch[0].replace(/\W/g, '')
+        site = getSiteById(siteId)
+    }
+    return site
+}
+
+const getSiteById = (id) => {
+    const {
+        app: {sites}
+    } = getConfig()
+    if (!sites.length) return undefined
+    if (!id) throw new Error('id is required')
+
+    return sites.find((site) => site.id === id)
+}
+
+const getSiteByAlias = (alias) => {
+    const {
+        app: {sites}
+    } = getConfig()
+    if (!sites.length) return undefined
+    if (!alias) throw new Error('alias is required')
+
+    return sites.find((site) => site.alias === alias)
+}
+/**
+ * A util to create RegExp for siteId and siteAlias based on site configuration from pwa-kit-config.json
+ *
+ * @example
+ * sites: [{id: 'RefArchGlobal', alias: 'global'}]
+ *
+ * getSitesRegExp()
+ * // returns {
+ *     siteIdsRegExp: /=RefArchGlobal\b|/RefArch/gi
+ *     siteAliasRegExp: /=global\b|/global/gi
+ * }
+ *
+ * @returns {object}
+ */
+export const getSitesRegExp = () => {
+    const {
+        app: {sites}
+    } = getConfig()
+    let idsRegExpPattern = []
+    let aliasRegExpPattern = []
+
+    sites.forEach((site) => {
+        idsRegExpPattern.push(`(/${site.id}/)`)
+        idsRegExpPattern.push(`(=${site.id}\\b)`)
```

Comment:
What is the `\b` about?

Reply:
It is to mark the beginning or end of a word. This is to avoid matching the partial string of a site id. We want the whole string (for example, when we look for RefArch, but don't want the `RefArch` in the `RefArchGlobal`